edb/ir/ast.py

```diff
@@ -413,6 +413,9 @@ class Param:
     name: str
     """Parameter name"""
 
+    optional: bool
```

Comment:
Let's use `required: bool` instead for consistency with pointers and the cardinality inference machinery.

Reply:
Fixed. Also fixed `irast.Parameter(optional)` for consistency (but we can revert that commit if you wish).